### PR TITLE
Better ability handling in the subclass drawer.

### PR DESCRIPTION
**Pull request by ryan-rushton**
This creates a more natural handling of abilities in the subclass drawer. Abilities will be single select  as per the game. Mods remain the same.

https://user-images.githubusercontent.com/7344652/146637344-00a61b05-7efc-438d-8293-1653f75fd0bb.mp4

### CODE REVIEW

File: src/app/loadout-drawer/subclass-drawer/SubclassPlugDrawer.tsx [hunk 1]
```diff
@@ -186,10 +180,15 @@ function getPlugsForSubclass(
...
 
         if (plugSetHash && profileResponse) {
-          const plugsWithMaxSelectable: PlugsWithMaxSelectable = {
+          const plugSet: PlugSetWithDefaultPlug = {
             plugs: [],
             plugSetHash,
             maxSelectable: socketGroup.length,
+            defaultPlug: defs.InventoryItem.get(
+              firstSocket.socketDefinition.singleInitialItemHash
+            ) as PluggableInventoryItemDefinition,
```
Comment (bhollis) [inline src/app/loadout-drawer/subclass-drawer/SubclassPlugDrawer.tsx:189]: Do we need to use a special function to get the default?

File: src/app/loadout/plug-drawer/PlugSection.tsx [hunk 2]
```diff
@@ -9,40 +9,52 @@ import SelectablePlug from './SelectablePlug';
...
-export interface PlugsWithMaxSelectable {
+export interface PlugSet {
   /** The hash that links to the PlugSet definition. */
   plugSetHash: number;
   /** A list of plugs from this plugset. */
   plugs: PluggableInventoryItemDefinition[];
   /** The maximum number of plugs a user can select from this plug set. */
   maxSelectable: number;
+  /**
+   * The select behaveour of the plug set.
```
Comment (bhollis) [inline src/app/loadout/plug-drawer/PlugSection.tsx:22]:
```suggestion
   * The select behavior of the plug set.
```

File: src/app/loadout/plug-drawer/PlugSection.tsx [hunk 3]
```diff
@@ -9,40 +9,52 @@ import SelectablePlug from './SelectablePlug';
  * a list of plugs, plus some metadata about:
  * - the maximum we should let the user choose at once
  * - the plugset whence these plugs originate
+ * - the behaveour we use for selecting plugs
```
Comment (bhollis) [inline src/app/loadout/plug-drawer/PlugSection.tsx:12]:
```suggestion
 * - the behavior we use for selecting plugs
```

**Review by bhollis — APPROVED** [review]
Looks mostly like a rename so it's hard to see what the actual changes were, but I trust it works correctly.

File: src/app/loadout-drawer/subclass-drawer/SubclassPlugDrawer.tsx (hunk 1 above)
Comment (ryan-rushton) [reply src/app/loadout-drawer/subclass-drawer/SubclassPlugDrawer.tsx:189]: Yeah I should, its mainly used for armour mods due to the way artificer armour works but it wouldn't hurt to use it here for future proofing.

File: src/app/loadout/plug-drawer/PlugSection.tsx [hunk 4]
```diff
@@ -9,40 +9,52 @@ import SelectablePlug from './SelectablePlug';
...
   /** A list of plugs from this plugset. */
   plugs: PluggableInventoryItemDefinition[];
   /** The maximum number of plugs a user can select from this plug set. */
   maxSelectable: number;
+  /**
+   * The select behavior of the plug set.
+   * multi: how armour mods are selected in game, you need to manually remove ones that have been added.
+   * single: how abilities in subclasses are selected, selecting an option replaces the current one.
+   */
+  selectionType: 'multi' | 'single';
```
Comment (ryan-rushton) [inline src/app/loadout/plug-drawer/PlugSection.tsx:26]: Just to call it out, the main change was the addition of this thing. It allows us to determine the behaviour or the items when you select a new one. I.e. whether it adds a new one (multi) or it replaces the current one (single).

**Review by bhollis — APPROVED** [review]

**Review by nev-r — APPROVED** [review]
this works well as intended but it feels like i am chasing.
image

clicking seems to turn the other one on. i really feel like the lit-upness is inverted...
this is fine for now but maybe we should reconsider diming as an indicator of selectedness

**Comment by bhollis** [comment]
It'd be helpful if we didn't dim the orange border for the selected item.

**Comment by ryan-rushton** [comment]
Yeah I can do that, I was a bit undecided on showing the currently selected one as opaque because I normally associate that with being disabled. Should we never show things being dimmed for abilities and maybe just have the border?

**Comment by bhollis** [comment]
Well it's communicating two different things - what's selected, and what *can* be selected.

**Pull request merged**